docs/branching-strategy.md

```diff
@@ -0,0 +1,98 @@
+> * `main`
+> * `develop`
+> 
+> When clicking merge on a PR to one of these branches, it is your duty to ensure that PRs are raised to merge that branch into all downstream supported base branches.
+
+## Summary of approach
+
+### The base branches
+
+We use a variant of `git-flow`, where there are three types of base branches: the `main`, `develop`, and `release/*` branches.
+
+* The `main` branch is the public-facing base branch and **represents the last official release**. It's also used for docs.
+* The `develop` branch is the primary integration branch, for work targeting the next protocol version.
+* The `release/*` branches are for all named protocol versions (i.e. each 1.X in the naming scheme. Typically patch releases should re-use same branch). A subset of release branches are **currently supported** - typically these are those currently on a live environment, or under development.
+
+At any given time, there is a strict ordering on the supported base branches, where the process aims to guarantee all work on previous branches is in the downstream branches. This order (from most upstream to most downstream) is as follows:
+
+* Released `release/*` branches still compatibile with a mainnet network
+* `main`
+* Unreleased `release/*` branches
+* `develop`
+
+The latest ordering is at the top of this document.
+
+### Development process
+
+When working on changes:
+* You will first need to select the correct base branch to create your feature branch from. For some epics, it is acceptable to choose a long running `feature/*` or `epic/*` branch as your base, to break up the work into separate reviews.
+* Your branch should start `feature/*`, or variants on naming such as `hotfix/*`, `tweak/*`, `docs/*` are permitted. The specific name should be prefixed by a JIRA ticket or Github issue reference where appropriate, e.g. `feature/NODE-123-my-feature` for JIRA tickets or `feature/gh-1235-my-feature` for github issues.
+* When you raise a PR, you will need to ensure you select the appropriate base branch before creating the PR. **The default, `main` is typically not correct!**
+
+> [!IMPORTANT]
+> 
+> Finally, when a PR is merged, it is **the PR merger's responsibility** to ensure that the _base branch_ that was merged into is then merged into _all downstream base branches_ (ideally one by one, like a waterfall).
+>
+> If there is a merge conflict, this should be handled by creating a special `conflict/X-into-Y-DATE` branch (for branches `X`, `Y` and `DATE`) from `X`, and putting in a PR with a merge target of `Y`.
+>
+> But if this process is properly followed, such merge conflicts will be rare. 
+
+## Which base branch should I use for X?
+
+### Code changes
+
+Features branches usually branch from `develop`, unless they need to target the current/previous protocol version, in which case, they will need to target the appropriate `release/*` branch.
+
+### Doc changes
+
+Docs changes should use a base branch of `main`.
```

Comment:
even if they describe a change that is being merged e.g. to develop?

_(maybe I just don't understand what "doc" here means)_

Reply:
Good point - have updated to clarify, thanks! :D